js/binance.js

```diff
@@ -7690,14 +7691,91 @@ module.exports = class binance extends Exchange {
         return this.parseOpenInterests (response, symbol, since, limit);
     }
 
+    async fetchOpenInterest (symbol, params = {}) {
+        /**
+         * @method
+         * @name binance#fetchOpenInterest
+         * @description retrieves the open interest of a contract trading pair
+         * @see https://binance-docs.github.io/apidocs/futures/en/#open-interest
+         * @see https://binance-docs.github.io/apidocs/delivery/en/#open-interest
+         * @see https://binance-docs.github.io/apidocs/voptions/en/#open-interest
+         * @param {string} symbol unified CCXT market symbol
+         * @param {object} params exchange specific parameters
+         * @returns {object} an open interest structure{@link https://docs.ccxt.com/en/latest/manual.html#interest-history-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {};
+        if (market['option']) {
+            const optionParts = symbol.split ('-');
```

Comment:
@Dan-krm we don't need to do this you can normalize the symbol or use its properties